cvat/apps/consensus/merge_consensus_jobs.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (C) 2024 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from typing import Dict, List, Tuple, Union
+
+import datumaro as dm
+import django_rq
+from django.conf import settings
+from django.db import transaction
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+
+from cvat.apps.consensus.consensus_reports import (
+    ComparisonReport,
+    generate_job_consensus_report,
+    generate_task_consensus_report,
+    save_report,
+)
+from cvat.apps.consensus.models import ConsensusSettings
+
+# from datumaro.components.operations import IntersectMerge
+from cvat.apps.consensus.new_intersect_merge import IntersectMerge
+from cvat.apps.dataset_manager.bindings import import_dm_annotations
+from cvat.apps.dataset_manager.task import PatchAction, patch_job_data
+from cvat.apps.engine.models import Job, JobType, StageChoice, StateChoice, Task, User
+from cvat.apps.engine.utils import (
+    define_dependent_job,
+    get_rq_job_meta,
+    get_rq_lock_by_user,
+    process_failed_job,
+)
+from cvat.apps.quality_control.quality_reports import JobDataProvider
+
+
+def get_consensus_jobs(task_id: int) -> Tuple[Dict[int, List[Tuple[int, User]]], List[Job]]:
+    jobs = {}  # parent_job_id -> [(consensus_job_id, assignee)]
+
+    for job in Job.objects.select_related("segment").filter(
+        segment__task_id=task_id, type=JobType.CONSENSUS.value
+    ):
+        assert job.parent_job_id
+
+        # if the job is in NEW state, it means that the job isn't annotated
+        if job.state == StateChoice.NEW.value:
+            continue
+        jobs.setdefault(job.parent_job_id, []).append((job.id, job.assignee))
+
+    parent_job_ids = list(jobs.keys())
+
+    parent_jobs: List[Job] = []
+    # remove parent jobs that are not in annotation stage
+    for parent_job_id in parent_job_ids:
+        parent_job = Job.objects.filter(id=parent_job_id, type=JobType.ANNOTATION.value).first()
+
+        if parent_job.stage == StageChoice.ANNOTATION.value:
+            parent_jobs.append(parent_job)
+        else:
+            jobs.pop(parent_job_id)
+
+    return jobs, parent_jobs
```

Comment:
```suggestion
    job_map = {}  # parent_job_id -> [(consensus_job_id, assignee)]
    parent_jobs: dict[int, Job] = {}
    for job in Job.objects.prefetch_related("segment", "parent_job").filter(
        segment__task_id=task_id, 
        type=JobType.CONSENSUS.value, 
        parent_job__stage=StageChoice.ANNOTATION.value,
        parent_job__isnull=False,
    ).exclude(state=StateChoice.NEW.value):
        job_map.setdefault(job.parent_job_id, []).append((job.id, job.assignee))
        parent_jobs.setdefault(job.parent_job_id, dob.parent_job)

    return job_map, list(parent_jobs.values())
```

Reply:
It's `parent_job_id` not `parent_job`. So instead of `job.parent_job` should I use Job.prefetch multiple times for each `parent_job_id`?

Reply:
There is no such relation as `parent_job_id`. It's an automatic field from Django in the `job` objects. It's stored in the job table, so fetched together with other job fields. Parent job is not a column, it's a relation, so it's fetched in `prefetch_related()`.

Reply:
In the model, it's defined as a `parent_job_id` itself. That's why when I replaced the mentioned code with this snippet it failed with the same message.

Reply:
Well, it's better define the field as a foreign key to the same model, an example is here: https://github.com/cvat-ai/cvat/blob/a69e1228ac8ab120ca9f1274348e83ddcb89d4be/cvat/apps/engine/models.py#L818

Reply:
Ah, on doing so it prompts that the object of type Job isn't JSON serializable...